insights/parsers/systemd/unitfiles.py

```diff
@@ -81,3 +93,76 @@ def is_on(self, service_name):
             bool: True if service is enabled, False otherwise
         """
         return self.services.get(service_name, False)
+
+
+@parser('systemctl_list-units')
+class ListUnits(Parser):
+
+    """
+    The ListUnits class parses the output of ``/bin/systemctl list-units`` and provides
+    information about all the services listed under it.
+
+    Output of Command::
+
+        sockets.target                      loaded active active    Sockets
+        swap.target                         loaded active active    Swap
+        systemd-shutdownd.socket            loaded active listening Delayed Shutdown Socket
+        neutron-dhcp-agent.service          loaded active running   OpenStack Neutron DHCP Agent
+        neutron-openvswitch-agent.service   loaded active running   OpenStack Neutron Open vSwitch Agent
+
+    Example:
+
+        >>> conf = shared[ListUnits]
+        >>> conf.get_service_details('swap.target')
+        {'LOAD': 'loaded', 'ACTIVE': 'active', 'SUB': 'active', 'UNIT': 'swap.target'}
+        >>> conf.unit_list['swap.target']
+        {'LOAD': 'loaded', 'ACTIVE': 'active', 'SUB': 'active', 'UNIT': 'swap.target'}
+        >>> conf.unit_list['random.service']
+        {'LOAD': None, 'ACTIVE': None, 'SUB': None, 'UNIT': None}
+    """
+    def __init__(self, *args, **kwargs):
+        self.unit_list = {}
+        """dict: Dictionary service detail like active, running, exited, dead"""
+        super(ListUnits, self).__init__(*args, **kwargs)
+
+    def parse_content(self, content):
+        """
+        Main parsing class method which stores all interesting data from the content.
+
+        Args:
+            content (context.content): Parser context content
+        """
+        # 'static' means 'on' to fulfill dependency of something else that is on
+        # man systemctl - "is-enabled" knows these states
+        valid_states = {'invalid', 'loaded', 'inactive', 'active',
+                        'exited', 'running', 'failed', 'mounted', 'waiting', 'plugged'}
+
+        valid_units = {'service', 'socket', 'device', 'mount', 'automount', 'swap', 'target',
+                       'path', 'timer', 'slice', 'scope'}
+
+        for line in get_active_lines(content):
+            parts = line.split(None)  # AWK like split, strips whitespaces
+            if any(part in valid_states for part in parts):
+                service_details = {}
+                service_details['LOAD'] = parts[1]
+                service_details['ACTIVE'] = parts[2]
+                service_details['SUB'] = parts[3]
+                if any(unit in parts[0] for unit in valid_units):
+                    service_details['UNIT'] = parts[0]
+                    self.unit_list[parts[0]] = service_details
+
+    def get_service_details(self, service_name):
```

Comment:
I'd really like to see 'is_loaded', 'is_active' and 'is_running' methods, which would do things like:
```
def is_loaded(self, service_name)
    """ Is the service loaded? """
    return self.unit_list.get(service_name, 'unloaded') == 'loaded'
```
These helper methods make accessing the actual state we're interested in much easier.  But I'm happy with this as is - we can add those later.